adlfs/spec.py

```diff
@@ -460,7 +458,10 @@ def __init__(
         self.location_mode = location_mode
         self.credential = credential
         self.request_session = request_session
-        self.socket_timeout = socket_timeout
+        if socket_timeout is not _SOCKET_TIMEOUT_DEFAULT:
+            warnings.warn(
+                "socket_timeout is deprecated and has no effect.", DeprecationWarning
```

Comment:
A FutureWarning will be visible in more places than the DeprecationWarning.

```suggestion
                "socket_timeout is deprecated and has no effect.", FutureWarning
```